tests/run-pass/stack_overflow_crash.rs

```diff
@@ -0,0 +1,50 @@
+extern crate rayon;
+
+use rayon::*;
+
+use std::process::{self, Command};
+use std::env;
+
+#[cfg(target_os = "linux")]
+use std::os::unix::process::ExitStatusExt;
+
+
+fn force_stack_overflow(depth: u32) {
+    let buffer = [0u8; 1024*1024];
+    if depth > 0 {
+        force_stack_overflow(depth - 1);
+    }
+}
+
+fn main() {
+    if env::args().len() == 1 {
+        // first check that the recursivecall actually causes a stack overflow, and does not get optimized away
+        {
+            let status = Command::new(env::current_exe().unwrap())
+                .arg("8")
+                .status()
+                .unwrap();
+            assert_eq!(status.code(), None);
+            #[cfg(target_os = "linux")]
+            assert_eq!(status.signal(), Some(6));
```

Comment:
How about `libc::SIGABRT`?  Same value, but more clear about what we're looking for.